rust/migrate-wicked/src/reader.rs

```diff
@@ -1,15 +1,16 @@
 use crate::interface::{Interface, ParentKind};
-use quick_xml::de::from_str;
+use crate::MIGRATION_SETTINGS;
 use regex::Regex;
 use std::collections::HashMap;
 use std::fs::{self, read_dir};
 use std::path::{Path, PathBuf};
 
-pub fn read_xml(str: &str) -> Result<Vec<Interface>, quick_xml::DeError> {
-    from_str(replace_colons(str).as_str())
+pub struct InterfacesResult {
+    pub interfaces: Vec<Interface>,
+    pub error: Option<anyhow::Error>,
```

Comment:
WDYT?

```suggestion
    pub warnings: Vec<anyhow::Error>,
```

Reply:
WRT the Vector, everything apart from unhandled fields is a critical error, so it can currently only ever be one. I don't really see a usecase for multiple here and if there is one in the future it shouldn't be hard to implement then. Or do you see something that would warrant multiple warnings here?
WRT the renaming, sure.

Reply:
I actually just compared the patterns of the other `XXXResult` structs.